### PR TITLE
DDF-2824-Security-Hardening

### PR DESCRIPTION
**Pull request by bakejeyner**
#### What does this PR do?
Updates some of the security hardening documentation.
#### Who is reviewing it? 
@brjeter 
@blen-desta 
#### Choose 2 committers to review/merge the PR. 
@coyotesqrl
@ricklarsen - Documentation
@stustison 
#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc [hunk 1]
```diff
@@ -19,27 +19,21 @@ Even if a user claim is more restrictive, the guest claim will grant access, so
...
-* *{hardening-step}*
-
-Uninstall the <<_guest_interceptor,Guest Interceptor>>.
-
-==== Configuring Guest Validator if Prohibiting Guest Users
-
-Uninstall the <<_guest_validator,Guest Validator>>.
+[NOTE]
+====
+If using the included Idp for authentication disable the `Allow Guest Access` option by <<_configuring_the_idp_server, Configuring the Idp Server>>.
```
Comment (brjeter) [inline distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc]: ✏️ put a `,` after "authentication"

File: distribution/docs/src/main/resources/content/_configuring/configuring-idp.adoc [hunk 2]
```diff
@@ -19,17 +19,22 @@ Install the included IdP implementation.
...
-.. file URL (file:), or
-.. XML block to refer to desired metadata, e.g. (https://localhost:8993/services/saml/sso/metadata)
+. Configure Authentication Request requirements
+.. Disable the *Require Signed AuthnRequests* option to allow processing of authentication requests without signatures.
+. Configure Guest Access:
+.. Disable the *Allow Guest Access* option to disallow a user to authenticate against the Idp with a guest account.
+. Configure SP Metadata:
+.. Click the `+` next to SP Metadata to add a new entry.
+.. Populate the new entry:
+... with an HTTPS URL (https://),
```
Comment (blen-desta) [inline distribution/docs/src/main/resources/content/_configuring/configuring-idp.adoc]: ✏️  
Populate the new entry *with*:
- an HTTPS URL (https://),
- *a* file URL (file:), or
- *an* XML block to refer to desired metadata

File: distribution/docs/src/main/resources/content/_configuring/configuring-sts.adoc [hunk 3]
```diff
@@ -77,6 +77,6 @@ Be sure to limit the hosts that are allowed to connect to the STS:
 * *{hardening-step}*
 
 * Open the `<${branding}_HOME>/etc/system.properties` file.
-* Edit the line `ws-security.subject.cert.constraints = .*`.
-** Remove the `.*` and replace with a comma-delimited list of regular expressions for the DNs of all desired hosts (`<MY_HOST_DN>`):
-*** `ws-security.subject.cert.constraints = <MY_HOST_DN>,<OTHER_HOST_DN>`
+* Edit the line `ws-security.subject.cert.constraints = .\*CN=<MY_HOST_CN>.*`.
+** By default this will only allow your hostname. To allow other desired hosts add their CN's to the regular expression within parenthesis deliminated by `|`:
```
Comment (blen-desta) [inline distribution/docs/src/main/resources/content/_configuring/configuring-sts.adoc]: I think it's `CNs` not `CN's`

File: distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc (hunk 1 above)
Comment (coyotesqrl) [inline distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc]: s/b `IdP` whenever it isn't used in code.

File: distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc [hunk 4]
```diff
@@ -19,27 +19,21 @@ Even if a user claim is more restrictive, the guest claim will grant access, so
...
+====
 
 === Allowing Guest User Access
 
 Guest authentication must be enabled and configured to allow guest users.
 Once the guest user is configured, redaction and filtering of metadata is done for the guest user the same way it is done for normal users.
 
-To enable guest authentication for a context, use the <<_web_context_policy_manager, Web Context Policy Manager>> configuration to change the *Authentication Type* for that context to *Guest*.
+To enable guest authentication for a context, use the <<_configuring_the_web_context_policy_manager, Web Context Policy Manager>> configuration to change the *Authentication Type* for that context to *Guest*.
```
[the remark this answers is not in the document]
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc]: +1

File: distribution/docs/src/main/resources/content/_configuring/configuring-idp.adoc [hunk 5]
```diff
@@ -19,17 +19,22 @@ Install the included IdP implementation.
...
-. Populate the new entry:
-.. with an HTTPS URL (https://),
-.. file URL (file:), or
-.. XML block to refer to desired metadata, e.g. (https://localhost:8993/services/saml/sso/metadata)
+. Configure Authentication Request requirements
+.. Disable the *Require Signed AuthnRequests* option to allow processing of authentication requests without signatures.
+. Configure Guest Access:
+.. Disable the *Allow Guest Access* option to disallow a user to authenticate against the Idp with a guest account.
+. Configure SP Metadata:
+.. Click the `+` next to SP Metadata to add a new entry.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/content/_configuring/configuring-idp.adoc]: ✏️ change `click` to Select

File: distribution/docs/src/main/resources/content/_configuring/configuring-sts.adoc [hunk 6]
```diff
@@ -77,6 +77,6 @@ Be sure to limit the hosts that are allowed to connect to the STS:
 * *{hardening-step}*
 
 * Open the `<${branding}_HOME>/etc/system.properties` file.
-* Edit the line `ws-security.subject.cert.constraints = .*`.
-** Remove the `.*` and replace with a comma-delimited list of regular expressions for the DNs of all desired hosts (`<MY_HOST_DN>`):
-*** `ws-security.subject.cert.constraints = <MY_HOST_DN>,<OTHER_HOST_DN>`
+* Edit the line `ws-security.subject.cert.constraints = .\*CN=<MY_HOST_CN>.*`.
+** By default this will only allow your hostname. To allow other desired hosts add their CNs to the regular expression within parenthesis deliminated by `|`:
```
Comment (coyotesqrl) [inline distribution/docs/src/main/resources/content/_configuring/configuring-sts.adoc]: s/b "parentheses" (plural)

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**